DuckDuckGo/OmniBar.swift

```diff
@@ -171,37 +183,48 @@ class OmniBar: UIView {
         }
     }
     
-    @IBAction func onClearButtonPresed(_ sender: Any) {
-        textField.text = ""
+    @IBAction func onClearButtonPressed(_ sender: Any) {
+        clear()
+        refreshState(state.onTextClearedState)
+    }
+    
+    @IBAction func onBookmarksButtonPressed(_ sender: Any) {
+        omniDelegate?.onBookmarksPressed()
     }
     
     @IBAction func onMenuButtonPressed(_ sender: UIButton) {
         omniDelegate?.onMenuPressed()
     }
     
-    @IBAction func onContentBlockerButtonPressed(_ sender: UIButton) {
+    @IBAction func onContentBlockerButtonPressed(_ sender: Any) {
         omniDelegate?.onContenBlockerPressed()
     }
 }
 
 extension OmniBar: UITextFieldDelegate {
     
     func textFieldDidBeginEditing(_ textField: UITextField) {
-        onEditingMode()
+        refreshState(state.onEditingStartedState)
+        DispatchQueue.main.async {
```

Comment:
Do you mean the async? We're in the middle of starting editing and the ui stutters if we block to highlight.